pymc3/distributions/continuous.py

```diff
@@ -723,42 +686,69 @@ def logp(self, value):
+        print(lower.eval())
+        norm = _truncated_normal_normalization(mu, sigma, lower, upper)
+        logp = Normal.logp(value, mu=mu, sigma=sigma) - norm
         bounds = [sigma > 0]
-        if self.lower_check is not None:
-            bounds.append(value >= self.lower)
-        if self.upper_check is not None:
-            bounds.append(value <= self.upper)
+        if lower_check is not None:
+            bounds.append(value >= lower)
+        if upper_check is not None:
+            bounds.append(value <= upper)
+        print([e.eval() for e in bounds])
         return bound(logp, *bounds)
 
-    def _normalization(self):
-        mu, sigma = self.mu, self.sigma
-
-        if self.lower_check is None and self.upper_check is None:
-            return 0.0
-
-        if self.lower_check is not None and self.upper_check is not None:
-            lcdf_a = normal_lcdf(mu, sigma, self.lower)
-            lcdf_b = normal_lcdf(mu, sigma, self.upper)
-            lsf_a = normal_lccdf(mu, sigma, self.lower)
-            lsf_b = normal_lccdf(mu, sigma, self.upper)
-
-            return at.switch(self.lower > 0, logdiffexp(lsf_a, lsf_b), logdiffexp(lcdf_b, lcdf_a))
-
-        if self.lower_check is not None:
-            return normal_lccdf(mu, sigma, self.lower)
-        else:
-            return normal_lcdf(mu, sigma, self.upper)
-
     def _distr_parameters_for_repr(self):
         return ["mu", "sigma", "lower", "upper"]
 
 
+def _truncated_normal_prepare_mu_sigma_params(mu, sigma, tau, sd):
+    sigma = sd if sd is not None else sigma
+    tau, sigma = get_tau_sigma(tau=tau, sigma=sigma)
+    sigma = at.as_tensor_variable(sigma)
+    tau = at.as_tensor_variable(tau)
+    mu = at.as_tensor_variable(floatX(mu))
+    assert_negative_support(sigma, "sigma", "TruncatedNormal")
+    assert_negative_support(tau, "tau", "TruncatedNormal")
+    return mu, sigma
+
+
+def _truncated_normal_normalization(mu, sigma, lower, upper):
+    lower, lower_check, upper, upper_check = _truncated_normal_prepare_lower_and_upper(lower, upper)
+    if lower_check is None and upper_check is None:
+        return 0.0
+
+    if lower_check is not None and upper_check is not None:
+        lcdf_a = normal_lcdf(mu, sigma, lower)
+        lcdf_b = normal_lcdf(mu, sigma, upper)
+        lsf_a = normal_lccdf(mu, sigma, lower)
+        lsf_b = normal_lccdf(mu, sigma, upper)
+        return at.switch(lower > 0, logdiffexp(lsf_a, lsf_b), logdiffexp(lcdf_b, lcdf_a))
+    if lower_check is not None:
+        return normal_lccdf(mu, sigma, lower)
+    else:
+        return normal_lcdf(mu, sigma, upper)
+
+
+def _truncated_normal_prepare_lower_and_upper(lower, upper):
+    lower_check = at.as_tensor_variable(floatX(lower)) if lower is not None else lower
+    upper_check = at.as_tensor_variable(floatX(upper)) if upper is not None else upper
+
+    lower = (
+        at.as_tensor_variable(floatX(lower))
+        if lower is not None
+        else at.as_tensor_variable(-np.inf)
```

Comment:
Are these `infs` used anywhere?

Reply:
There were a couple of tests for the cases in which either `upper` or `lower` were None. ie `TestTruncatedNormalLower` and `TestTruncatedNormalUpper`

Reply:
It seems to me they are not used, because the logp changes based on this and doesn't include the `None` extremes anyway. I am not even sure we need to separate between `lower` and `lower_check`.